### PR TITLE
fix(components/modals): ensure confirmation modals include accessibility labels

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
[AB#2340557](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2340557)

### CODE REVIEW

File: libs/components/modals/src/lib/modules/modal/modal.component.ts [hunk 1]
```diff
@@ -58,13 +58,29 @@ export class SkyModalComponent implements AfterViewInit, OnDestroy {
   @Input()
   public tiledBody: boolean | undefined;
 
-  public ariaDescribedBy: string;
+  /**
+   * @internal
+   */
+  @Input()
+  public set ariaDescribedBy(id: string | undefined) {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/modals/src/lib/modules/modal/modal.component.ts:65]: I needed to be able to pass in `undefined` here so  I modified this to use a setter and then went ahead and changed `ariaLabelledBy` as well just for consistency.

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/888/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/888/storybooks/ag-grid-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/888/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/888/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/888/storybooks/pages-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/888/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/888/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/888/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/888/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/888/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/888/playground/)

File: libs/components/modals/src/lib/modules/confirm/confirm.component.spec.ts [hunk 2]
```diff
@@ -341,4 +341,228 @@ describe('Confirm component', () => {
...
+            text: 'baz',
+            action: 'baz',
+            autofocus: true,
+          },
+        ],
+      });
+
+      fixture.detectChanges();
+
+      const buttons = fixture.nativeElement.querySelectorAll(
```
Comment (Blackbaud-SteveBrush) [inline libs/components/modals/src/lib/modules/confirm/confirm.component.spec.ts]: There's a lot of repeated code in these tests. Maybe you could create a utility function that could be shared by all?
```
async function verifyAccessibility(options: {...}): Promise<void> {
  const fixture = createConfirm(options);

  fixture.detectChanges();

  const buttons = fixture.nativeElement.querySelectorAll(
    '.sky-confirm-buttons .sky-btn'
  );

  await expectAsync(fixture.nativeElement).toBeAccessible();
  
  buttons[0].click();
}
```
And then each test can call...
```
await verifyAccessibility({
  message: 'confirm message',
  body: 'additional text',
  type: SkyConfirmType.Custom,
  buttons: [
    {
      text: 'Custom label',
      action: 'foo',
    },
  ],
});
```

File: libs/components/modals/src/lib/modules/confirm/confirm.component.ts [hunk 3]
```diff
@@ -55,6 +58,8 @@ export class SkyConfirmComponent {
 
     this.message = config.message;
     this.body = config.body;
+    // Using the service here instead of the directive due to the body element being conditional and thus a template variable being unavailable at an outer scope
```
Comment (Blackbaud-SteveBrush) [inline libs/components/modals/src/lib/modules/confirm/confirm.component.ts]: Can we reword to `...due to the confirm component's "body" container being conditional...`? I was initially confused that this comment was talking about the `<body>` element.

Comment (Blackbaud-TrevorBurch) [reply libs/components/modals/src/lib/modules/confirm/confirm.component.ts]: Sure. Done

File: libs/components/modals/src/lib/modules/confirm/confirm.component.spec.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/modals/src/lib/modules/confirm/confirm.component.spec.ts]: Good call. Done

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**